blockprod/src/detail/mod.rs

```diff
@@ -180,19 +180,51 @@ impl BlockProduction {
         &self,
         current_tip: Id<GenBlock>,
         min_block_timestamp: BlockTimestamp,
+        transactions: Vec<SignedTransaction>,
+        transaction_ids: Vec<Id<Transaction>>,
+        include_mempool: bool,
```

Comment:
Yep, I think this is the issue I'm facing right now in the functional tests. Ok, I'll rethink this.

Reply:
I think the problem of relative ordering of transactions given by hex vs given by ID is easily solved by introducing an enum that is either a transaction or an id, and having one vector with the enum as values.

How will these interleave with transactions from mempool is a different question. One possibility is as now: mempool transactions go at the end. If the user wants to include a transaction that depends on a mempool transaction, the IDs of its dependencies have to be explicitly listed. Would it be good enough, at least for now?

Reply:
>  If the user wants to include a transaction that depends on a mempool transaction

hmm... unless this was before I started changing things and I'm getting confused, including transaction hex followed by transaction Ids (so that they're pulled from the mempool) that had inputs from those transaction hexes didn't work. From the error message, I think this was because when I inserted the transactions into the mempool they were going into the orphan pool and so `generate_block()` was failing.

So is that above scenario supposed to work?

> the IDs of its dependencies have to be explicitly listed

Yep, that's what I was thinking given we can assume that the caller knows what they're doing because they're overriding the defaults.

Reply:
We also need to be careful that the transaction that's passed in by the user as ID or in hex is not included again from mempool. That would lead to an invalid block due to double spending